index.js

```diff
@@ -156,26 +178,65 @@ const ScrollableTabView = React.createClass({
          }}
          ref={(scrollView) => { this.scrollView = scrollView; }}
          {...this.props.contentProps}>
-         {this._children().map((child, idx) => {
-           return <View
-             key={child.props.tabLabel + '_' + idx}
-             style={{width: this.state.containerWidth, }}>
-             {child}
-           </View>;
-         })}
+           {scenes}
         </ViewPagerAndroid>
       );
     }
   },
 
+  _composeScenes() {
+    let scenes = [];
+    this._children().forEach((child, idx) => {
+      let key = this._makeSceneKey(child, idx);
+      let selected = false;
+
+      // if key doesnt exist(typically before each tab is first mounted) then put a dummy child instead of real
+      // this ensures # of pages remain the same , leaving scrolling and click tab effect to be handled as intended
+      if(!this._keyExists(this.state.sceneKeys, key)) {
+        let scene = (
+          <SceneComponent
+          key={key}
+          style={{width: this.state.containerWidth, }}
+          selected={selected}>
+            <View tabLabel={child.props.tabLabel}/>
+          </SceneComponent>
+        );
+        scenes.push(scene);
+      }
+      // if keyexists and the tab is viewed for the first time then SceneComponent is mounted, else its updated and rendered
+      // Also when a tab is selected,StaticContainer inside the SceneComponent makes sure that only tab being viewed is rendered
+      if(this._keyExists(this.state.sceneKeys, key)) {
+        if(idx === this.state.currentPage) selected = true;
+        let scene = (
+          <SceneComponent
```

Comment:
This component and component above are almost the same? Maybe would be better to keep 1 component and inside would be ```this._keyExists(this.state.sceneKeys, key) ? {<View tabLabel={child.props.tabLabel}/> : child}```. And this forEach we will be able to replace with map